### ISSUE TITLE
Issues with pipes inside of backpacks inside a table

### ISSUE THREAD

**Issue by skyzyx** [github/cmark#24]
**Actual Result:**

This used to work as expected.

| Parameter     | Type           | Description |
| ------------- | ---------------|------------ |
| `id`          | `integer|null` | The identifier for the `approval` resource, if available. |
| `description` | `string`       | The description for the `approval` resource, if available. |

**Expected Result:**

| Parameter     | Type           | Description |
| ------------- | ---------------|------------ |
| `id`          | `integer\|null` | The identifier for the `approval` resource, if available. |
| `description` | `string`       | The description for the `approval` resource, if available. |

**Comment by kivikakk** [comment]
This is now intended behaviour — we revised the spec when it became apparent that there were some nasty edge cases with trying to correctly detect the pipe `|` outside a span (such as backticks) when there were multiple adjacent emphasis markers. You can see some of the history at #18, #19 and #21.

**Comment by Crissov** [comment]
This issue has been [discussed in the Commonmark Discourse](https://talk.commonmark.org/t/parsing-strategy-for-tables/2027) as well.  

As a user, I almost always don't want vertical bars `|` to trump backticks `` ` `` &ndash; nothing should trump them and you cannot even backslash-escape anything else (including a backtick) within a backtick span ` \` `. 
As a spec writer and implementer, I understand the benefits of clearly separating block and inline parsing.  The question, though, is whether table cells are blocks or inlines.

**Comment by kivikakk** [comment]
:wave: @Crissov 

As a user, I didn't want that either — that's why I went to quite so much effort to make it work correctly. Like any developer, too, I wasn't really happy about throwing that away in #21 for this "solution", but unfortunately when confronted with situations like in #18, it turned out to be unsustainable; the edge cases get weirder and weirder, and trying to cover them up (#19) just makes for an even harder to explain system, the edge cases get even stranger, and the code starts to become a nightmare. (We ended up parsing inlines three times for each row there, iirc.)

Simply put, there's not a great, consistent way to do this. I think it does make sense to consider the table cell structure at the block level, inasmuch as they're not spanned by delimiters but rather separated by them. It's easiest to consider the entire table structure one large block which in parsed in its entirety, before inline parsing then happens, which is also literally the case.

**Comment by Crissov** [comment]
As I said, I understand it from a developer's POV. Anyway, wouldn't it work to parse for inline code spans without linebreaks first, then blocks and finally inlines?
